modules/baremetal/main.tf

```diff
@@ -2,17 +2,39 @@ data "ibm_is_subnet" "subnet" {
   identifier = var.subnet_id
 }
 
+resource "ibm_is_virtual_network_interface" "bms" {
+  count           = length(var.allowed_vlan_ids) > 0 ? 1 : 0
+  name            = "${var.name}-vni"
+  subnet          = var.subnet_id
+  resource_group  = var.resource_group_id
+  security_groups = var.security_group_ids
+}
+
+# Secondary VNI
+resource "ibm_is_virtual_network_interface" "bms_secondary" {
+  count           = var.secondary_vni_enabled ? 1 : 0
+  name            = "${var.name}-secondary-vni"
+  subnet          = var.secondary_subnet_id != "" ? var.secondary_subnet_id : var.subnet_id
+  resource_group  = var.resource_group_id
+  security_groups = var.secondary_security_group_ids != null ? var.secondary_security_group_ids : var.security_group_ids
+}
+
 resource "ibm_is_bare_metal_server" "bms" {
-  profile        = var.profile
-  name           = var.name
-  image          = var.image_id
-  keys           = var.ssh_key_ids
-  vpc            = data.ibm_is_subnet.subnet.vpc
-  zone           = data.ibm_is_subnet.subnet.zone
-  bandwidth      = var.bandwidth
-  access_tags    = var.access_tags
-  resource_group = var.resource_group_id
-  user_data      = var.user_data
+  profile            = var.profile
+  name               = var.name
+  image              = var.image_id
+  keys               = var.ssh_key_ids
+  vpc                = data.ibm_is_subnet.subnet.vpc
+  zone               = data.ibm_is_subnet.subnet.zone
+  bandwidth          = var.bandwidth
+  access_tags        = var.access_tags
+  resource_group     = var.resource_group_id
+  user_data          = var.user_data
+  enable_secure_boot = var.enable_secure_boot
+
+  trusted_platform_module {
```

Comment:
Just checking that this block doesn't need to be dynamic? It can be included always? I don't think the new tpm_mode is restricted from being null, so you may want to make sure that no matter what, tpm_mode is valid by the time of this non-dynamic block.